backend/allocation/graphql/query/courseAllocation.py

```diff
@@ -0,0 +1,56 @@
+
+import graphene
+from allocation.graphql.types.allocation import CourseAndFacultyType,AllocationFilterInput
+from course.graphql.types.course import CourseType
+from allocation.models import CourseAllocation
+from user.models import Faculty
+from user.graphql.types.user import FacultyType
+from backend.api import APIException
+from backend.api.decorator import login_required
+
+class AllocationQueries(graphene.ObjectType):
+    all_allocation = graphene.List(CourseAndFacultyType,)
+    # batch_wise_allocation = graphene.List(CourseAndFacultyType
+    allocation = graphene.List(
+        CourseAndFacultyType,
+        filter = graphene.Argument(AllocationFilterInput,required=True))
+    
+    @login_required
+    def resolve_allocation(self,info,filter):
```

Comment:
Remove this filter argument as that is not needed because we can filter the data on the frontend,
add (year, is_even_sem) as the argument to this query (preference_sem_identifier) and filter out allocations in that time period. add the decoraters `@resolve_user` `@staff_privilege_required` to this query. Because only staff should be able to call this.